File: Facebook/src/logics/adSetController.js
const facebookAdsService = require('../services/facebookAdsService');

const createAdSet = async (req, res) => {
  try {
    // Extract ad set data from the request body
    const { name, campaignId, targeting } = req.body;

    // Call the Facebook Ads service to create the ad set
    const adSet = await facebookAdsService.createAdSet(name, campaignId, targeting);

    res.status(201).json(adSet);
  } catch (error) {
    console.error('Error creating ad set:', error);
    res.sendStatus(500);
  }
};

module.exports = {
  createAdSet,
};

File: FacebookSrv/controllers/adset_controller.py
from flask import request, jsonify
from services.facebook_ads_service import create_ad_set

def create_ad_set_controller():
    data = request.get_json()
    name = data.get('name')
    campaign_id = data.get('campaign_id')
    targeting = data.get('targeting')

    ad_set = create_ad_set(name, campaign_id, targeting)
    return jsonify(ad_set), 201
